src/Http/Http.php

```diff
@@ -857,75 +731,80 @@ public function run(Request $request, Response $response, string $context): stat
                     if (in_array('*', $error->getGroups())) {
-                        self::setResource('error', function () use ($e) {
-                            return $e;
-                        }, [], $context);
-                        \call_user_func_array($error->getAction(), $this->getArguments($error, $context, [], $request->getParams()));
+                        $dependency = new Dependency();
+                        $this->container->set(
+                            $dependency
+                                ->setName('error')
+                                ->setCallback(fn () => $e)
+                        )
+                        ;
+
+                        $this->prepare($context, $error, [], $request->getParams())->inject($error, true);
                     }
                 }
             }
 
             return $this;
         }
 
-        if (null === $route && null !== self::$wildcardRoute) {
-            $route = self::$wildcardRoute;
-            $this->route = $route;
-            $path = \parse_url($request->getURI(), PHP_URL_PATH);
-            $route->path($path);
-
-            self::setResource('route', fn () => $route, [], $context);
-        }
-
         if (null !== $route) {
-            return $this->execute($route, $request, $context);
+            return $this->lifecycle($route, $request, $context);
         } elseif (self::REQUEST_METHOD_OPTIONS == $method) {
             try {
                 foreach ($groups as $group) {
                     foreach (self::$options as $option) { // Group options hooks
                         if (in_array($group, $option->getGroups())) {
-                            \call_user_func_array($option->getAction(), $this->getArguments($option, $context, [], $request->getParams()));
+                            $this->prepare($context, $option, [], $request->getParams())->inject($option, true);
                         }
                     }
                 }
 
                 foreach (self::$options as $option) { // Global options hooks
                     if (in_array('*', $option->getGroups())) {
-                        \call_user_func_array($option->getAction(), $this->getArguments($option, $context, [], $request->getParams()));
+                        $this->prepare($context, $option, [], $request->getParams())->inject($option, true);
                     }
                 }
             } catch (\Throwable $e) {
                 foreach (self::$errors as $error) { // Global error hooks
                     if (in_array('*', $error->getGroups())) {
-                        self::setResource('error', function () use ($e) {
-                            return $e;
-                        }, [], $context);
-                        \call_user_func_array($error->getAction(), $this->getArguments($error, $context, [], $request->getParams()));
+                        $dependency = new Dependency();
+                        $this->container->set(
```

Comment:
shouldn't this error be set on `context` ?

Reply:
Not sure I follow, which error?